tools/lsp/preview/native.rs

```diff
@@ -236,3 +250,67 @@ pub fn send_message_to_lsp(message: PreviewToLspMessage) {
     };
     sender.send_message_to_lsp(message);
 }
+
+// This function overrides the default app menu and makes the "Quit" item merely hide the UI,
+// as the life-cycle of this process is determined by the editor. The returned menuitem must
+// be kept alive for the duration of the event loop, as otherwise muda crashes.
```

Comment:
To clarify: I tried to reproduce this merely with the coda example, but I haven't succeeded quit yet. There's some native dangling pointer that's somewhere. But I've spent the entire morning on this and I'd prefer to leave it like this.

Evenentually I'd like to remove this code altogether and rather have a callback in the Slint API that queries if it's okay to quit the event loop. Then the LSP can decline, hide the window, and move on.